src/helpers/calculationStepFunctions/numericCalculationFunctions.test.js

```diff
@@ -0,0 +1,38 @@
+import { calculateValue } from "./numericCalculationFunctions";
+
+describe("NumericCalculationTests", () => {
+    test('QuestionIdentifier has .', () => {
+      expect(calculateValue(
+        "(([Q1]*[Q1.2])[Q2])-[Q3]",
+        [{"date":"2023-07-19T11:14:18.377-04:00","stepIdentifier":"Q1","resultIdentifier":"Q1","answers":["3"]},{"date":"2023-07-19T11:14:18.944-04:00","stepIdentifier":"Form1","resultIdentifier":"Q1.2","answers":["4"]},{"date":"2023-07-19T11:14:20.338-04:00","stepIdentifier":"Q2","resultIdentifier":"Q2","answers":["2"]},{"date":"2023-07-19T11:14:20.849-04:00","stepIdentifier":"Q3","resultIdentifier":"Q3","answers":["3"]}]
+      )).toBe(`Calculation Error - QuestionIdentifier "Q1.2" cannot have a "."`);
+    });
+
+    test('Invalid Variable', () => {
+      expect(calculateValue(
+        "(([Q8]*[Q1_2])+[Q2])-[Q3]",
+        [{"date":"2023-07-19T11:14:18.377-04:00","stepIdentifier":"Q1","resultIdentifier":"Q1","answers":["3"]},{"date":"2023-07-19T11:14:18.944-04:00","stepIdentifier":"Form1","resultIdentifier":"Q1_2","answers":["4"]},{"date":"2023-07-19T11:14:20.338-04:00","stepIdentifier":"Q2","resultIdentifier":"Q2","answers":["2"]},{"date":"2023-07-19T11:14:20.849-04:00","stepIdentifier":"Q3","resultIdentifier":"Q3","answers":["3"]}]
+      )).toBe('Calculation Error - Unmatched Variable');
+    });
+
+    test('Checkbox - multiselect', () => {
+      expect(calculateValue(
+        "(([Q1]*[Q1_2])+[Q2])-[Q3]",
+        [{"date":"2023-07-19T11:14:18.377-04:00","stepIdentifier":"Q1","resultIdentifier":"Q1","answers":["3", "4"]},{"date":"2023-07-19T11:14:18.944-04:00","stepIdentifier":"Form1","resultIdentifier":"Q1_2","answers":["4"]},{"date":"2023-07-19T11:14:20.338-04:00","stepIdentifier":"Q2","resultIdentifier":"Q2","answers":["2"]},{"date":"2023-07-19T11:14:20.849-04:00","stepIdentifier":"Q3","resultIdentifier":"Q3","answers":["3"]}]
+      )).toBe('Calculation Error - Calculation not available for multiple-select text choice questions (Q1)');
+    });
+
+    test('Calculation includes variable that is not a number', () => {
+      expect(calculateValue(
+        "(([Q1]*[Q1_2])+[Q2])-[Q3]",
+        [{"date":"2023-07-19T11:14:18.377-04:00","stepIdentifier":"Q1","resultIdentifier":"Q1","answers":["Jacob"]},{"date":"2023-07-19T11:14:18.944-04:00","stepIdentifier":"Form1","resultIdentifier":"Q1_2","answers":["4"]},{"date":"2023-07-19T11:14:20.338-04:00","stepIdentifier":"Q2","resultIdentifier":"Q2","answers":["2"]},{"date":"2023-07-19T11:14:20.849-04:00","stepIdentifier":"Q3","resultIdentifier":"Q3","answers":["3"]}]
+      )).toBe('Calculation Error - Answer not a number (Q1)');
+    });
+
+    test('Calculation Passes', () => {
+      expect(calculateValue(
+        "(([Q1]*[Q1_2])+[Q2])-[Q3]",
```

Comment:
Maybe I am missing it but it seems like we did not do anything special for the ```[]``` syntax this to work, no?  So that is all built into Math.js and actually you do not even need the [] in most cases.  Q1 * Q3 works fine as an expression it seems.

However, in my quick test, seems like Q1_2 does not work and you need the brackets for some reason.  Curious if you know why or have figured out what the [] actually does

I think there are other characters that do not work as intended; for instance spaces - [Q1 2] doesn't work.

Reply:
The [] is the array operator in mathjs. In this case, it doesn't really do anything. I used it because I liked having a "syntax" to designate which is a variable. 

When I was testing the Q1_2 without the [] worked correctly and gave the correct response, but you're right. The [Q1 2] and the Q1 2, will actually calculate but result in an correct calculation. 

Here is a [list of operators ](https://mathjs.org/docs/expressions/syntax.html), so many of these won't work. Doesn't really make sense to error check for all of them. But what is our list of basic ones? dash, space, period? what about parentheses, brackets?